code/modules/admin/holder2.dm

```diff
@@ -145,6 +145,29 @@ you will have to do something like if(client.holder.rights & R_ADMIN) yourself.
 		return 1
 	return 0
 
+/**
+ * Requires the holder to have all the rights specified
+ *
+ * rights_required = R_ADMIN|R_EVENT means they must have both flags, or it will return false
+ */
+/proc/check_rights_all(rights_required, show_msg = TRUE, mob/user = usr)
```

Comment:
This proc seems unnecessary, the `check_rights` already works with multiple rights.

Reply:
it will check if they have either `R_ADMIN` or `R_EVENT`. This proc makes sure they have both. I could also just do `check_rights(R_ADMIN, FALSE, user) && check_rights(R_EVENT, FALSE, user)` in tons more places, but I think thats a lot more messy and harder to read.

Reply:
Surely `check_rights(R_ADMIN+R_EVENT)` works? Since youll then be checking the bitmask for 2 bits exists?

Reply:
basically the way check_rights works is it takes the mask `(R_ADMIN+R_EVENT)` or if you were considering them boolean masks, `(R_ADMIN|R_EVENT)` (this is usually how its written for check_rights). It then performs a boolean and between the mask, and the rights of our admin holder, and if that mask produced any result at all, then it returns true.

check_rights_all does the masking, and then compares it to the original rights given, and only if theyre the same does it return true.